domainservice/src/main/kotlin/systems/ajax/malov/stockanalyzer/kafka/processor/StockPriceNotificationProcessor.kt

```diff
@@ -1,79 +1,36 @@
 package systems.ajax.malov.stockanalyzer.kafka.processor
 
 import jakarta.annotation.PostConstruct
-import org.apache.kafka.clients.consumer.ConsumerRecord
-import org.apache.kafka.clients.producer.ProducerRecord
-import org.bson.types.ObjectId
 import org.springframework.stereotype.Component
 import reactor.core.scheduler.Schedulers
 import reactor.kafka.receiver.KafkaReceiver
-import reactor.kafka.sender.KafkaSender
-import reactor.kafka.sender.SenderRecord
-import reactor.kotlin.core.publisher.toFlux
-import systems.ajax.malov.internalapi.KafkaTopic
-import systems.ajax.malov.internalapi.output.pubsub.stock.notification_stock_price.proto.NotificationStockPrice
+import reactor.kotlin.core.publisher.toMono
 import systems.ajax.malov.internalapi.output.pubsub.stock.stock_price.proto.StockPrice
-import systems.ajax.malov.stockanalyzer.entity.MongoUserTrackedSymbol
-import systems.ajax.malov.stockanalyzer.mapper.proto.BigDecimalProtoMapper.convertBigDecimalProtoToBigDecimal
-import systems.ajax.malov.stockanalyzer.repository.UserTrackedSymbolRepository
+import systems.ajax.malov.stockanalyzer.kafka.producer.StockPriceNotificationProducer
+import systems.ajax.malov.stockanalyzer.service.NotificationStockPriceService
 
 @Component
 class StockPriceNotificationProcessor(
     private val kafkaStockPriceConsumer: KafkaReceiver<String, StockPrice>,
-    private val kafkaNotificationStockPriceSender: KafkaSender<String, NotificationStockPrice>,
-    private val userTrackedSymbolRepository: UserTrackedSymbolRepository,
+    private val kafkaNotificationStockPriceSender: StockPriceNotificationProducer,
+    private val notificationStockPriceService: NotificationStockPriceService,
 ) {
     @PostConstruct
     fun subscribeToEvents() {
         kafkaStockPriceConsumer.receiveAutoAck()
             .flatMap { stockPriceConsumerRecords ->
                 stockPriceConsumerRecords.flatMap { stockPriceConsumerRecord ->
-                    createUsersNotifications(stockPriceConsumerRecord)
+                    notificationStockPriceService.createUsersNotifications(stockPriceConsumerRecord.value())
                 }
-            }.flatMap {
-                kafkaNotificationStockPriceSender.send(it.toFlux())
+            }.flatMap { notifications ->
+                kafkaNotificationStockPriceSender.sendNotificationStockPrice(notifications.map { it._1 })
+                    .thenMany(notifications.toMono())
+            }
+            .flatMap { notifications ->
+                notificationStockPriceService
+                    .deleteUserTrackedSymbols(notifications.map { it._2 })
             }
             .subscribeOn(Schedulers.boundedElastic())
             .subscribe()
```

Comment:
here kafka consumer logic is mixed with your domain logic. please separate them so your consuming code will look similar to this

`        kafkaStockPriceConsumer.receiveAutoAck()
            .flatMap { records ->
                records.flatMap { handle(it.value()) }
            }
            .subscribeOn(Schedulers.boundedElastic())
            .subscribe()`